### DISCUSSION TITLE
3.4 - Beta

### DISCUSSION THREAD

**Discussion by vikdevelop** [vikdevelop/SaveDesktop#303] [Announcements]
# [Release notes](https://github.com/vikdevelop/SaveDesktop/releases/)

I would love for you to take part in testing this new version! If you run into any issues, feel free to report them via GitHub Issues or join the conversation in GitHub Discussions by clicking the "Join Discussion" button. There, you'll also find [instructions on how to set up syncing using the cloud folder](https://github.com/vikdevelop/SaveDesktop/discussions/303#discussioncomment-10255434).

#### 1. Flatpak
Add the Flathub Beta repository to your system:
```
flatpak remote-add --if-not-exists flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
```
Install the SaveDesktop application from this repository:

```
flatpak install flathub-beta io.github.vikdevelop.SaveDesktop
```
#### 2. Snap
To install this release, run the following command:
```
snap install savedesktop --beta
```

<hr /><em>This discussion was created from the release <a href='https://github.com/vikdevelop/SaveDesktop/releases/'>3.4 - Beta</a>.</em>

**Comment by vikdevelop** [comment]
# Sync using a cloud storage folder
## Requirements
- Create folders on the computers you want to synchronize with each other that will synchronize with your cloud storage (e.g., Google Drive or Next Cloud), which can be achieved using GNOME Online Accounts or rclone.

## Settings in SaveDesktop
On the first computer:

1. Open the SaveDesktop app
2. On the Sync page, click on the "Set up the sync file" button and then on the "Change" button
3. Click on the "Periodic saving" and select the folder that is synchronized with your cloud storage as a periodic saving folder
4. If the periodic saving file does not exist, click on the Create button

On the second computer:

1. Open the SaveDesktop app
2. Go to the Sync page and click the "Connect to the cloud storage" button.
3. Click on the "Select cloud drive folder" button and select the folder that is synced with the same cloud storage as the first computer.
4. Select the periodic synchronization interval, because if you leave that to Never, the synchronization doesn't work.
5. Bidirectional synchronization is enabled by default, but if you only want to synchronize in one direction, e.g. from the first computer to the second computer, turn it off.

**Reply by 13hannes11** [reply]
I seem to have trouble setting it up properly. To my understanding the sync only works one way. So I did as described. What is missing for me is the ability to test the import for the sync.

Unclear for me:

1. Have I set the sync up correctly and how do I verify
2. Flatpak apps do not seem to sync?
3. Are password protected files supported for this sync mode?

**Reply by vikdevelop** [reply]
> I seem to have trouble setting it up properly. To my understanding the sync only works one way. So I did as described. What is missing for me is the ability to test the import for the sync.
> 
> Unclear for me:
> 
> 1. Have I set the sync up correctly and how do I verify
> 2. Flatpak apps do not seem to sync?
> 3. Are password protected files supported for this sync mode?

Yes, the synchronization in this case only works in one direction so far.
1. To check whether the sync is working correctly or not, type in the terminal `flatpak run io.github.vikdevelop.SaveDesktop --sync`. If you have Never selected for periodic synchronization, synchronization will not work (as before).
2. If you have Flatpak applications and their data in your configuration archive, then yes, but they will not install until you log back in
3. Encrypted archives are unfortunately not yet supported in either periodic saving mode or sync mode. 

    You can't even select those encrypted archives, because in this case properly you would have to select the folder on the "source" machine that you have synchronized with your cloud storage as the periodic saving folder where the periodic saving file will be saved (which also creates a SaveDesktop-sync-file text file that contains the name of the archive because Flatpak permissions do not reflect that the file was overwritten).
    
    On the "receiving" computer, you then also select the folder that you have synced with your cloud storage and in which that periodic saving file is.

I hope I've explained this clearly.

**Reply by 13hannes11** [reply]
The explanation makes a lot of sense. I think I managed to set it up correctly now. Generally this seems to work and I will continue to test it in the next days.

Probably the user feedback and workflow for setting up sync can be a bit improved but otherwise it is a very useful tool!

**Reply by vikdevelop** [reply]
So, what's your experience with syncing, does it work for you?

Otherwise, bidirectional sync should also work, just make sure to select the same folder on the source computer that is synced with your cloud storage. Also, ensure that the file name on the other computer matches the one on the source computer. If the names are different, just set the same file name format in the periodic saving section. In short, follow the same procedure as on the first computer.

**Reply by 13hannes11** [reply]
The syncing itself works but it seems to not sync flatpak applications properly. I installed Apostrophe via Flatpak on the machine that is periodically saving but it has not been installed on the machine reading from file. I have not had time to take a closer look.

**Comment by vikdevelop** [comment]
I released the 3.4-beta-07 version, which is the last beta version before releasing the stable version, and their release notes are below:
- Fixed a bug with auto-mounting the cloud drive folders when the folder mounts after the synchronization, so this is why the synchronization didn't work
- In the synchronization mode, the Flatpak apps and their data will be installed immediately after the finished synchronization, so it is not necessary to log out of the system
- New strings have been released for the application itself and also for the documentation related to the new release, so they can already be translated via Weblate

What else would you like to see fixed or slightly improved before the stable release?

**Comment by vikdevelop** [comment]
As no further bugs have been reported or discovered, I am releasing a stable version 3.4, which will be available for download on Flathub in a few hours, or in Snap or the native version in a few moments.
